_pages/fsa/dynamicforms/core/basic-form-validations.md

```diff
@@ -0,0 +1,285 @@
+        },
+        {
+            "value": "shouldBeGreater"
+        }
+        ]
+    }
+],
+```
+- **maxValue** - Default Angular validator
+- **minValue** - Default Angular validator
+- **maxLength** - Default Angular validator
+- **minLength** - Default Angular validator
+- **number** - Triggers error message is input type is not number.
+- **compareDates** - Compares input date to the value of other field. Accepts two arguments:
+  - name of the field to compare (string)
+  - operator (string) can be 'shouldBeGreater' or 'shouldBeLess'
+
+JSON example, compare input to the value of the 'vehiclePurchaseDate' field:
+```typescript
+"validations": [
+    {
+        "name": "compareDates",
+        "arguments": [
+        {
+            "value": "vehiclePurchaseDate"
+        },
+        {
+            "value": "shouldBeGreater"
+        }
+        ]
+    }
+],
+```
+- **checkValue** - Checks if input data matches any item from array of allowed values. Accepts one argument (array of strings)
+JSON example, field is valid only in case string "no" is added to the input:
+```typescript
+"validations": [
+    {
+    "name": "checkValue",
+    "arguments": [
+        {
+        "value": [
+            "no"
+        ]
+        }
+    ]
+    }
+]
+```
+- **containsValue** - Checks if input value contains string form argument array. Accepts one argument (array of strings) 
+JSON example:
+```typescript
+"validations": [
+    {
+    "name": "containsValue",
+    "arguments": [
+        {
+        "value": ["test string"]
+        }
+    ]
+    }
+]       
+```
+- **compareNumbers** - Compares input value to other field. Accepts two arguments:
+  - name of the field to compare (string)
+  - operator (string) can be 'shouldBeGreater' or 'shouldBeLess' 
+   
+JSON example:
+```typescript
+"validations": [
+    {
+        "name": "compareNumbers",
+        "arguments": [
+        {
+            "value": "propertyValue"
+        },
+        {
+            "value": "shouldBeLess"
+        }
+        ]
+    }
+],
+```
+- **email** - Checks if input data is valid email
+- **alphanumeric** - Allows only numbers and letters as input data 
+
+## How all this works
+
+Before Form gets rendered, all validations from JSON are processed and in case they have corresponding implementation defined  in config they are transferred to mapped **ValidationFn**. Name from validation object corresponds the name defined in form configuration on SPA application. In this particular case it will be validation with name compareToCurrentDate :
+
+```typescript
+DynamicFormsConfig
+validators: {
+  ....
+  compareToCurrentDate: {
+    validator:  DefaultFormValidators.compareToCurrentDate
+  },
+  ....
+}
+```
+
+All arguments defined in JSON file represents inputs parameters for compareToCurrentDate function from DefaultFormValidators and they are added to the function in run-time. In our example based on mapping defined in configuration we will find following validation function:
```

Comment:
"All arguments defined in JSON file **represents**" => "All arguments defined in JSON file **represent**" - singular